src/osa/scripts/gainsel_webmaker.py

```diff
@@ -0,0 +1,127 @@
+from pathlib import Path
+from astropy.table import Table
+from datetime import datetime, timedelta
+from osa.utils.utils import date_to_dir, date_to_iso
+from osa.configs.config import cfg
+from osa.paths import DEFAULT_CFG
+from osa.nightsummary.nightsummary import run_summary_table
+import numpy as np
+import pandas as pd
+from argparse import ArgumentParser
+from osa.configs import options
+from osa.scripts.sequencer_webmaker import html_content
+
+def valid_date(string):
+    """Check if the string is a valid date and return a datetime object."""
+    return datetime.strptime(string, "%Y%m%d")
+
+
+common_parser = ArgumentParser(add_help=False)
+common_parser.add_argument(
+    "-c",
+    "--config",
+    type=Path,
+    default=DEFAULT_CFG,
+    help="Use specific config file [default configs/sequencer.cfg]",
+)
+common_parser.add_argument(
+    "-d",
+    "--date",
+    help="Date (YYYYMMDD) of the start of the night",
+    type=valid_date,
+)
+
+
+def check_gainsel_jobs_runwise(date: str, run_id: int) -> bool:
+    """Search for failed jobs in the log directory."""
+    base_dir = Path(cfg.get("LST1", "BASE"))
+    log_dir = base_dir / f"R0G/log/{date}"
+    history_files = log_dir.glob(f"gain_selection_{run_id:05d}.????.history")
+
+    success_subruns = 0
+    failed_subruns = 0
+    pending_subruns = 0
+
+    for file in history_files:
+        if file.read_text() != "":
+            gainsel_rc = file.read_text().splitlines()[-1][-1]
+
+            if gainsel_rc == "1":
+                failed_subruns = failed_subruns+1
+
+            elif gainsel_rc == "0":
+                success_subruns = success_subruns+1
+        else:
+            pending_subruns = pending_subruns+1
+    return [pending_subruns, success_subruns, failed_subruns]
+
+def check_failed_jobs(date: str):
+    """Search for failed jobs in the log directory."""
+    summary_table = run_summary_table(datetime.fromisoformat(date))
+    data_runs = summary_table[summary_table["run_type"] == "DATA"]
+
+    gainsel_summary = []
+    for run in data_runs:
+        run_id = run["run_id"]
+        checkgainsel = check_gainsel_jobs_runwise(date.replace('-',''), run_id)
+        gainsel_summary.append([run_id, checkgainsel[0], checkgainsel[1], checkgainsel[2]])
```

Comment:
```suggestion
        gainsel_summary.append([run_id, gainsel_job_status])
```
I think that you should be able to append the whole returned list or tuple from that function without having to unpack the values

Reply:
Done!